### DISCUSSION TITLE
Convert macos-sonoma.sh to Sonoma.xml

### DISCUSSION THREAD

**Discussion by redleg** [quickemu-project/quickemu#1532] [Quickemu] [answered]
Just getting started with quickemu here, so pardon me if this has been asked and answered.

Assume I have a working MacOS Sonoma in macos-sonoma, with the corresponding macos-sonoma.sh

I would like to import this working VM into virt-manager.

Is there an automated way to convert the macos-sonoma.sh in macos-sonoma to a sonoma.xml file which I can relocate to /etc/libvirt/qemu and import into my existing virt-manager infrastructure with "virsh define sonoma.xml" ?

Thanks in advance......

**Comment by lj3954** [comment, marked as answer]
Quickemu does not currently support this feature.

**Comment by redleg** [comment]
OK, got it.

I got it working on my own, and once I have a working solution documented, I will post it here for others, in detail.  In the meantime:

I already had a working Ventura instance operating in virt-manager.

I used quickemu to install Sonoma.  I then copied and modified the XML file for my working Ventura instance for the Sonoma instance qcow2 image created with quickemu, relocated / renamed / relocated some files, and incorporated the modified XML file into libvirt / virt-manager.

Once inported in virt-manager, it's much easier to tweek.....                                  

End result, fully functional, i.e., able to access iServices like iMessage, Ventura and Sonoma VMs, and an installed but not-fully-functional (cannot access iServices)  Sequoia.  when I have this sorted to my satisfaction, and hopefuly the process automated to some degree, I'll post here.


Red

**Reply by ghost** [reply]
Hello @redleg, thanks for your work but I was wondering where I could find it? I have attempted this myself and I can't get it working so I would love to look at your solution. Thanks!

**Comment by danhertztech** [comment]
Hi @redleg , thanks for posting your solution in the future.   Be nice to have a quickemu .sh to virt-manager .xml export feature .

**Comment by redleg** [comment]
You are very welcome.

In the mean time, I found another project similar to quickemu.....  https://oneclick-macos-simple-kvm.notaperson535.is-a.dev

Click on Installation Tutorial, and then Manual Method.  The repo includes a script to generate am XML file and specific instructions on how to import it into virt-manager., along with A LOT of other interesting info which addresses some frequent issues.

It would be nice to see these two projects merge......

Red
